nsxt/resource_nsxt_policy_host_switch_profile.go

```diff
@@ -0,0 +1,557 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var policyHostSwitchProfileResourceType = []string{
+	model.PolicyBaseHostSwitchProfile_RESOURCE_TYPE_POLICYUPLINKHOSTSWITCHPROFILE,
+	model.PolicyBaseHostSwitchProfile_RESOURCE_TYPE_POLICYLLDPHOSTSWITCHPROFILE,
+	model.PolicyBaseHostSwitchProfile_RESOURCE_TYPE_POLICYNIOCPROFILE,
+	model.PolicyBaseHostSwitchProfile_RESOURCE_TYPE_POLICYEXTRACONFIGHOSTSWITCHPROFILE,
+	model.PolicyBaseHostSwitchProfile_RESOURCE_TYPE_POLICYVTEPHAHOSTSWITCHPROFILE,
+}
+
+var lacpLoadBalanceAlgorithms = []string{
+	model.Lag_LOAD_BALANCE_ALGORITHM_SRCMAC,
+	model.Lag_LOAD_BALANCE_ALGORITHM_DESTMAC,
+	model.Lag_LOAD_BALANCE_ALGORITHM_SRCDESTMAC,
+	model.Lag_LOAD_BALANCE_ALGORITHM_SRCDESTIPVLAN,
+	model.Lag_LOAD_BALANCE_ALGORITHM_SRCDESTMACIPPORT,
+}
+
+var lacpGroupModes = []string{
+	model.Lag_MODE_ACTIVE,
+	model.Lag_MODE_PASSIVE,
+}
+
+var lacpTimeoutTypes = []string{
+	model.Lag_TIMEOUT_TYPE_SLOW,
+	model.Lag_TIMEOUT_TYPE_FAST,
+}
+
+var uplinkTypes = []string{
+	model.Uplink_UPLINK_TYPE_PNIC,
+	model.Uplink_UPLINK_TYPE_LAG,
+}
+
+var teamingPolicies = []string{
+	model.TeamingPolicy_POLICY_FAILOVER_ORDER,
+	model.TeamingPolicy_POLICY_LOADBALANCE_SRCID,
+	model.TeamingPolicy_POLICY_LOADBALANCE_SRC_MAC,
+}
+
+var overlayEncapModes = []string{
+	model.PolicyUplinkHostSwitchProfile_OVERLAY_ENCAP_VXLAN,
+	model.PolicyUplinkHostSwitchProfile_OVERLAY_ENCAP_GENEVE,
+}
+
+func resourceNsxtPolicyHostSwitchProfile() *schema.Resource {
```

Comment:
I'd suggest to remove `policy` from the name, since its really a fabric resource

Reply:
Done.